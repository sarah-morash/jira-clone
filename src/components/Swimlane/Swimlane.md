<!-- prettier- ignore -->
```
const sampleTickets = [
  {
    id: "abc",
    description: "Some",
    estimatedTime: 20,
    loggedTime: 10
  },
  {
    id: "bc",
    description: "Some",
    estimatedTime: 20,
    loggedTime: 10
  }
];

<Swimlane title="Sample Title" tickets={sampleTickets} />
```
