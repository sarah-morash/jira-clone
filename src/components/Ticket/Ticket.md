<!-- prettier- ignore -->
Assigned
```
<Ticket
  id="KGSM-123"
  description="As a user I want to be able to see my components in the styleguide."
  assignedTo="K.G. OHearon"
  estimatedTime={20}
  loggedTime={10}
/>
```

Unassigned
```
<Ticket
  id="KGSM-123"
  description="Some Description"
  estimatedTime={20}
  loggedTime={10}
/>
```
