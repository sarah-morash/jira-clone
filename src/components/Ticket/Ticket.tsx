import { Card } from "@material-ui/core";
import * as React from "react";
import styled from "react-emotion";

export interface ITicket {
  /**
   * The Ticket ID
   */
  id: string;

  /**
   * The description of
   * the ticket
   */
  description: string;

  /**
   * The name of the person
   * assigned to a ticket
   */
  assignedTo?: string;

  /**
   * The original time estimate
   * of the ticket
   */
  estimatedTime: number;

  /**
   * The time logged against
   * this ticket
   */
  loggedTime: number;
}

const CardContentWrapper = styled.div`
  padding: 10px;
`;

const CardIdWrapper = styled.div`
  padding-bottom: 5px;
`;

const CardDescriptionWrapper = styled.div`
  padding: 5px 0;
`;

const CardMetaDataWrapper = styled.div`
  padding-top: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
export default (props: ITicket) => (
  <Card>
    <CardContentWrapper>
      <CardIdWrapper>{props.id}</CardIdWrapper>
      <CardDescriptionWrapper> {props.description} </CardDescriptionWrapper>
      <CardMetaDataWrapper>
        <div> {props.assignedTo ? props.assignedTo : "Unassigned"} </div>
        <div> {props.estimatedTime - props.loggedTime} </div>
      </CardMetaDataWrapper>
    </CardContentWrapper>
  </Card>
);
