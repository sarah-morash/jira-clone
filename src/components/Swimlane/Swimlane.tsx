import * as React from "react";
import styled from "react-emotion";
import Ticket, { ITicket } from "../Ticket/Ticket";

export interface ISwimlane {
  /**
   * Id
   */
  id: string;

  /**
   * Title
   */
  title: string;

  /**
   * Tickets
   */
  tickets: ITicket[];
}

const TicketWrapper = styled.div`
  &:nth-child(n + 2) {
    padding-top: 10px;
  }
`;

const SwimlaneWrapper = styled.div`
  background-color: #eeeeee;
  padding: 10px 5px;
`;

export default (props: ISwimlane) => (
  <SwimlaneWrapper>
    <div>{props.title.toUpperCase()}</div>
    {props.tickets.map((ticket: ITicket) => (
      <TicketWrapper key={ticket.id}>
        <Ticket {...ticket} />
      </TicketWrapper>
    ))}
  </SwimlaneWrapper>
);
