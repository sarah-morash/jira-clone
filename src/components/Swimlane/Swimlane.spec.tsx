import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Ticket, { ITicket } from "../Ticket/Ticket";
import Swimlane from "./Swimlane";

describe("Swimlane", () => {
  let wrapper: ShallowWrapper;
  const sampleTickets: ITicket[] = [
    {
      id: "abc",
      description: "Some",
      estimatedTime: 20,
      loggedTime: 10
    },
    {
      id: "bc",
      description: "Some",
      estimatedTime: 20,
      loggedTime: 10
    }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <Swimlane id="ABC" title="Sample Title" tickets={sampleTickets} />
    );
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Title
   */
  it("Should render its title in uppercase", () => {
    wrapper.setProps({ title: "swimlane title" });
    expect(wrapper.html()).toMatch("SWIMLANE TITLE");
  });

  /**
   * tickets
   */
  it("Should render each ticket passed to its tickets prop", () => {
    expect(wrapper.find(Ticket)).toHaveLength(2);
  });
});
