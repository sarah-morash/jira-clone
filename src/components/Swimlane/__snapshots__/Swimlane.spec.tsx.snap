// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swimlane Should match its snapshot 1`] = `
<SwimlaneWrapper>
  <div>
    SAMPLE TITLE
  </div>
  <TicketWrapper
    key="abc"
  >
    <Component
      description="Some"
      estimatedTime={20}
      id="abc"
      loggedTime={10}
    />
  </TicketWrapper>
  <TicketWrapper
    key="bc"
  >
    <Component
      description="Some"
      estimatedTime={20}
      id="bc"
      loggedTime={10}
    />
  </TicketWrapper>
</SwimlaneWrapper>
`;
